demo/src/main/java/com/example/EventHub/EventType/EventTypeController.java

```diff
@@ -28,4 +38,23 @@ public Iterable<EventType> allEventsTypes(){
         Iterable<EventType> allEventTypes = eventTypeRepository.findAll();
         return allEventTypes;
     }
+    @DeleteMapping("/delete")
+    public void deleteEventType(@RequestParam("id") Integer id){
+        Optional<EventType> foundEventType = eventTypeRepository.findById(id);
+        if(foundEventType.isPresent()){
+            eventTypeRepository.deleteById(id);
+        }
+    }
+
+    @PutMapping("/update")
+    public void updateEventType(@RequestParam("id") Integer id, @RequestBody EventTypeDTO eventTypeDTO, BindingResult bindingResult){
+        if (bindingResult.hasErrors()) {
+            throw new IllegalArgumentException();
+        } else {
+            EventType eventType = eventTypeMapper.toEntity(eventTypeDTO);
```

Comment:
This is not how you do it, in the frontend we have DTOs, not entities therefore we don't have id's therefore you would use @RequestParam("name") String name, then you use Even eventTypeToUpdate = eventTypeRepository.findByName(String name);
then you change the fields of  eventTypeToUpdate, then use eventTypeRepository.save(eventTypeToUpdate); because the id stays the sameas to when you extracted it from the db, the eventType would be update without a new row in the db being created.